teps/0143-concise-parameters-results.md

```diff
@@ -0,0 +1,848 @@
+      serviceAccountPath: release.json
+      versionTag: ${TEKTON_VERSION}
+      releaseBucket: gs://tekton-releases/pipeline
+  workspaces:
+    ...
+  pipelineRef:
+    resolver: git
+    params:
+      org: tektoncd
+      repo: pipeline
+      revision: ${TEKTON_RELEASE_GIT_SHA}
+      pathInRepo: tekton/release-pipeline.yaml
+---
+apiVersion: tekton.dev/v1
+kind: Pipeline
+metadata:
+  name: pipeline-release
+spec:
+  inputs:
+    params:
+      package:
+        description: package to release
+        default: github.com/tektoncd/pipeline
+      gitRevision:
+        description: the git revision to release
+      imageRegistry:
+        description: The target image registry
+        default: gcr.io
+      imageRegistryPath:
+        description: The path (project) in the image registry
+        default: tekton-releases
+      versionTag:
+        description: The X.Y.Z version that the artifacts should be tagged with
+      releaseBucket:
+        description: bucket where the release is stored. The bucket must be project specific.
+        default: gs://tekton-releases-nightly/pipeline
+      releaseAsLatest:
+        description: Whether to tag and publish this release as Pipelines' latest
+        default: "true"
+      buildPlatforms:
+        description: Platforms to build images for (e.g. linux/amd64,linux/arm64)
+        default: linux/amd64,linux/arm,linux/arm64,linux/s390x,linux/ppc64le
+      publishPlatforms:
+        description: |
+          Platforms to publish images for (e.g. linux/amd64,linux/arm64,windows/amd64). This
+          can differ from buildPlatforms due to the fact that a windows-compatible base image
+          is constructed for the publishing phase.
+        default: linux/amd64,linux/arm,linux/arm64,linux/s390x,linux/ppc64le,windows/amd64
+      serviceAccountPath:
+        description: The path to the service account file within the release-secret workspace
+  workspaces:
+    ...
+  results:
+    - name: commit-sha
+      description: the sha of the commit that was released
+      value: $(tasks.git-clone.results.commit)
+    - name: release-file
+      description: the URL of the release file
+      value: $(tasks.report-bucket.results.release)
+    - name: release-file-no-tag
+      description: the URL of the release file
+      value: $(tasks.report-bucket.results.release-no-tag)
+  tasks:
+    - name: git-clone
+      taskRef:
+        resolver: hub
+        params:
+          name: git-clone
+          version: "0.7"
+      workspaces:
+        ...
+      inputs:
+        params:
+          url: https://$(params.package)
+          revision: $(params.gitRevision)
+    - name: precheck
+      runAfter: [git-clone]
+      taskRef:
+        resolver: git
+        params:
+          repo: plumbing
+          org: tektoncd
+          revision: aeed19e5a36f335ebfdc4b96fa78d1ce5bb4f7b8
+          pathInRepo:  tekton/resources/release/base/prerelease_checks.yaml
```

Comment:
If changing the type to map, we need to validate the reserved names in the map key (`repo`, `org`, `revision`....)?

Reply:
Or we can think more about better syntax for resolver, but this is out of scope of this TEP

Reply:
I'm not sure we need to provide more validation than today, but I guess it could make sense to update the `ResolutionRequest` too. Would you mind providing more detail about this part?

Reply:
@Yongxuanzhang any comment on this?